### PR TITLE
Adding Time Hooks for measuring semantics performance

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
This PR introduces `hook_timer_start` and `hook_timer_stop`, which can be used to measure the execution time of an instruction or an entire semantics by getting the current timestamp when starting the timer and then when stopping it, we get the current timestamp again, and save the difference into a default file `hooks_time.txt` in the current directory.

### CODE REVIEW

**Review by theo25 — APPROVED** [review]
this looks good but given that I have contributed to this code, don't merge until @dwightguth approves as well.

**Review by dwightguth — APPROVED** [review]

**Pull request merged**
